examples/WORKSPACE

```diff
@@ -1,35 +1,56 @@
 workspace(name = "examples")
 
-# For released versions, use the tagged git-repository:
-# git_repository(
+# For released versions, use release tag:
+# http_repository(
 #     name = "io_grpc_grpc_java",
-#     remote = "https://github.com/grpc/grpc-java.git",
-#     tag = "<TAG>",
+#     sha256 = "<SHA>",
+#     strip_prefix = "grpc-java-<TAG>",
+#     url = "https://github.com/grpc/grpc-java/archive/<TAG>.zip",
 # )
 local_repository(
     name = "io_grpc_grpc_java",
     path = "..",
 )
 
-load("@io_grpc_grpc_java//:repositories.bzl", "grpc_java_repositories")
-
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-
-maven_jar(
-        name = "com_google_api_grpc_cloud_pubsub_v1",
-        artifact = "com.google.api.grpc:grpc-google-cloud-pubsub-v1:0.1.24",
-        sha1 = "601d8be0fd0cc0e050b1af3b88f191ada9a2f4e5",
+http_archive(
+    name = "rules_jvm_external",
+    sha256 = "62133c125bf4109dfd9d2af64830208356ce4ef8b165a6ef15bbff7460b35c3a",
+    strip_prefix = "rules_jvm_external-3.0",
+    url = "https://github.com/bazelbuild/rules_jvm_external/archive/3.0.zip",
 )
 
-maven_jar(
-        name = "com_google_api_grpc_proto_cloud_pubsub_v1",
-        artifact = "com.google.api.grpc:proto-google-cloud-pubsub-v1:0.1.24",
-        sha1 = "e6dd66635f674b4e380dfd3de252ae019a51a67e",
+load("@rules_jvm_external//:defs.bzl", "maven_install")
+load("@io_grpc_grpc_java//:repositories.bzl", "IO_GRPC_GRPC_JAVA_ARTIFACTS")
+load("@io_grpc_grpc_java//:repositories.bzl", "IO_GRPC_GRPC_JAVA_OVERRIDE_TARGETS")
+
+maven_install(
+    artifacts = [
+        "com.google.api.grpc:grpc-google-cloud-pubsub-v1:0.1.24",
+        "com.google.api.grpc:proto-google-cloud-pubsub-v1:0.1.24",
+    ] + IO_GRPC_GRPC_JAVA_ARTIFACTS,
+    generate_compat_repositories = True,
+    maven_install_json = "//:maven_install.json",
```

Comment:
Could this be too much for examples? It may add difficulty for users to copy the example and make changes. Even the examples of `rules_jvm_external` itself are not always pinning the maven artifacts. https://github.com/bazelbuild/rules_jvm_external/tree/master/examples

Reply:
Hmm... Good point. I'd rather have the examples be as "real" as possible, and I expect most Bazel users will want to pin. I've added a comment to mention how to regenerate the pinned versions.

I'm fine with dropping this if it becomes a problem, but I think I'd prefer to "do it real" first and remove it if it becomes a problem.